solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreakerManager.java

```diff
@@ -40,12 +41,15 @@
  * solution. There will be a follow up with a SIP for a schema API design.
  */
 public class CircuitBreakerManager implements PluginInfoInitialized {
+  private final SolrCore core;
```

Comment:
I am wary of adding a SolrCore dependency here since only one circuit breaker really uses it and this pollutes the API a fair bit.

Reply:
Please explain a different way to do it. The core is needed in order to look up the metric.

Reply:
Extend the manager class? Introdyxe a builder and have an implementation that takes the core?

We cannot increase the API surface for every new circuit breaker - - the manager has to be as minimal as possible

Reply:
Can we make `CircuitBreaker` subclasses that need the core implement `SolrCoreAware`? That would be the most elegant instead of having to pass all context down through the manager.

I'd also like to split up things even more to make CBs truly pluggable, and not managed by some common Manager that has to know about all plugins, and a common config class? But that's for another JIRA, and also a breaking change that is better suited for 9.0

Reply:
I was unaware of SolrCoreAware, I'll go check it out. Thanks.

The other larger change is to have circuit breakers only interrupt external requests and to not interrupt internal requests to shards. The current approach multiples failures.

Reply:
There should be some flag on the Threadpool I think, whether the request is internal (initiated by Solr) or external. Don't recall the details.

Reply:
> There should be some flag on the Threadpool I think, whether the request is internal (initiated by Solr) or external. Don't recall the details.

Via https://github.com/apache/solr/pull/112 I learnt about https://github.com/apache/lucene-solr/blob/releases/lucene-solr/8.8.2/solr/solrj/src/java/org/apache/solr/common/util/ExecutorUtil.java#L247-L256 i.e. `ExecutorUtil.(setServerThreadFlag|isSolrServerThread)`.

That may or may not be the flag you meant.

Reply:
For Solr 9, I am going to use the new flags introduced in SOLR-13528, which should be a cleaner way of achieving the goal (unless there is a problem I am missing, please?)

Reply:
> ... to make CBs truly pluggable, ... But that's for another JIRA ...

i've created a placeholder JIRA: https://issues.apache.org/jira/browse/SOLR-15474

Reply:
In my update to this PR we simply pass SolrCore to the constructor for now, as not all CBs need it.